### ISSUE TITLE
Evaluate Parallel Validation from upstream Bitcoin Unlimited

### ISSUE THREAD

**Issue by bitcartel** [zcash/zcash#2148]
Evaluate having a pool of worker threads process incoming blocks.  Upstream Bitcoin Unlimited has just merged in a working implementation:

> "Essentially Parallel Validation is a simple concept.  Rather than validating each block within the main processing thread, we instead create a separate thread to do the block validation. If more than one block arrives to be processed then we create yet another thread. There are currently up to 4 parallel block processing threads available making a big block DDOS attack impossible."

https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/254/files

https://github.com/BitcoinUnlimited/BUIP/blob/master/033.mediawiki

**Comment by daira** [comment]
I don't agree that this "[makes] a big block DDOS attack impossible", but it might be useful. How does it interact with existing parallel script validation (https://github.com/bitcoin/bitcoin/pull/2060)?
